pkg/templates/k8sresource_funcs.go

```diff
@@ -126,6 +126,38 @@ func (t *TemplateResolver) protect(value string) (string, error) {
 	return protectedPrefix + base64.StdEncoding.EncodeToString(encryptedValue), nil
 }
 
+// decrypt will decrypt a string that was encrypted using the protect method. An error is returned if the base64 or
```

Comment:
Optional comment :   If I were doing this , to keep it a little clean, I would have put protect function and all the encryption/decryption utils  functions in a separate file.   I initially meant for this file to be specific to funcs related to specific k8s resources.

Reply:
I think that is a good idea. I'll do it in a followup PR after this one since it'd also involve moving code that wasn't added in this PR.